tensorboard/data/experimental/experiment_from_dev.py

```diff
@@ -107,33 +116,41 @@ def get_scalars(self, runs_filter=None, tags_filter=None, pivot=None):
             )
             values.extend(list(response.points.values))
 
-        dataframe = pandas.DataFrame(
-            {
-                "run": runs,
-                "tag": tags,
-                "step": steps,
-                "wall_time": wall_times,
-                "value": values,
-            }
-        )
+        data = {
+            "run": runs,
+            "tag": tags,
+            "step": steps,
+            "value": values,
+        }
+        if include_wall_time:
+            data["wall_time"] = wall_times
+        dataframe = pandas.DataFrame(data)
         if pivot:
             dataframe = self._pivot_dataframe(dataframe)
         return dataframe
 
     def _pivot_dataframe(self, dataframe):
         num_missing_0 = np.count_nonzero(dataframe.isnull().values)
         dataframe = dataframe.pivot_table(
-            ["value", "wall_time"], ["run", "step"], "tag",
+            values=(
+                ["value", "wall_time"]
+                if "wall_time" in dataframe.columns
+                else "value"
+            ),
+            index=["run", "step"],
+            columns="tag",
         )
         num_missing_1 = np.count_nonzero(dataframe.isnull().values)
         if num_missing_1 > num_missing_0:
             raise ValueError(
-                "pivoted DataFrame contains %d missing value(s). "
+                "pivoted DataFrame contains missing value(s). "
                 "This is likely due to two timeseries having different "
                 "sets of steps in your experiment. "
                 "You can avoid this error by calling `get_scalars()` with "
                 "`pivot=False` to disable the DataFrame pivoting."
             )
+        dataframe = dataframe.reset_index()
```

Comment:
I’m having a hard time understanding what this actually does (and the
test doesn’t really help, because it just does the same thing as the
code). It seems confusing to me to pivot a table and then immediately
flatten the indices again. What kind of structure are we going for here?
I’ve read the Pandas docs but still don’t quite follow.